hazelcast/src/main/java/com/hazelcast/spi/impl/executionservice/InternalExecutionService.java

```diff
@@ -18,14 +18,27 @@
 
 import com.hazelcast.spi.ExecutionService;
 
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
 /**
  * The internal {@link ExecutionService}. Currently this method doesn't contains any additional methods, but
  * in the future they can be added here.
  */
 public interface InternalExecutionService extends ExecutionService {
 
+    ExecutorService getDurable(String name, boolean scheduled);
 
     void executeDurable(String name, Runnable command);
 
+    ScheduledFuture<?> scheduleDurable(String name, Runnable command, long delay, TimeUnit unit);
+
+    <V> ScheduledFuture<V> scheduleDurable(String name, Callable<V> command, long delay, TimeUnit unit);
+
+    ScheduledFuture<?> scheduleDurableWithRepetition(String name, Runnable command,
```

Comment:
Yes, this is a problem in the case of scheduling so frequent tasks, if we have more than one workers its no as much of a problem then since we skip execution if the task is running so we can drain the queue fast enough. But to be honest, if someone schedules a task to run so frequently that takes that much to complete, the queue size will be the least of his worries. However, I do see the point, so if we notice this in production we can address later.  WDYT ?